unittests/Sema/PlaceholderTypeInferenceTests.cpp

```diff
@@ -0,0 +1,98 @@
+//===--- PlaceholderTypeInferenceTests.cpp --------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "SemaFixture.h"
+#include "swift/Sema/ConstraintSystem.h"
+
+using namespace swift;
+using namespace swift::unittest;
+using namespace swift::constraints;
+
+TEST_F(SemaTest, TestPlaceholderInferenceForArrayLiteral) {
```

Comment:
I have to comment, because I really like how those tests were implemented
This is just great work @Jumhyn!

Reply:
All credit to whoever set up the testing infra here and made the constraint system so easy to spin up in a modular fashion like this!